libsplinter/src/mesh/mod.rs

```diff
@@ -76,28 +76,10 @@ use crate::mesh::reactor::Reactor;
 use crate::transport::Connection;
 
 /// Wrapper around payload to include connection id
-#[derive(Debug, Default, PartialEq)]
-pub struct InternalEnvelope {
-    id: usize,
-    payload: Vec<u8>,
-}
-
-impl InternalEnvelope {
-    pub fn new(id: usize, payload: Vec<u8>) -> Self {
-        InternalEnvelope { id, payload }
-    }
-
-    pub fn id(&self) -> usize {
-        self.id
-    }
-
-    pub fn payload(&self) -> &[u8] {
-        &self.payload
-    }
-
-    pub fn take_payload(self) -> Vec<u8> {
-        self.payload
-    }
+#[derive(Debug, PartialEq)]
+pub enum InternalEnvelope {
```

Comment:
Should this be pub? Also, with the changes where the signalling is via the `Control` struct, is this still needed to be an enum?

Reply:
I kept InternalEnvelope an enum so the recv would still get the shutdown notification. It only needs to be available in with the mesh mod but no where else. There are also a couple other components that are public but do not need to be. I added a commit to reduce their visibility to mesh.